### DISCUSSION TITLE
Can argo event serve business workflow?

### DISCUSSION THREAD

**Discussion by thanhpd-teko** [argoproj/argo-events#1165] [Q&A]
Hi, I'm using both argo workflow an argo events, and the are really great.
However, I wonder can argo serve some event and workflow for each user. For example:
- when new user register account => application publish an event to argo event => argo event trigger action send email to that user
- next, when that user take action1 => application publish an event to argo event => send another email to that user
- when that user take action2 => application publish an event to argo event => trigger send another email to user
- If that user has completed action1 and action2 then => trigger send congratulations to that user.
Is that possible with argo?

**Comment by thanhpd-teko** [comment]
@farcaller  please help.

**Reply by farcaller** [reply]
I'm not actually using argo anymore, sorry.

**Reply by thanhpd-teko** [reply]
Anyone help? :D

**Comment by louiscpalma** [comment]
I believe you're looking for `trigger conditions`

https://argoproj.github.io/argo-events/tutorials/06-trigger-conditions/
https://github.com/argoproj/argo-events/tree/master/examples/tutorials/06-trigger-conditions

**Reply by thanhpd-teko** [reply]
@louiscpalma Yep, I've read trigger conditions, but the point here is how to specify event for each user (Users register new acccount daily).

**Reply by thanhpd-teko** [reply]
For example:
If userA has completed action1 and action2 then plus point for that user
If userB has completed only action1, then still wait until he completed action2 then plus point for user B

**Reply by thanhpd-teko** [reply]
As I read here https://argoproj.github.io/argo-events/sensors/more-about-sensors-and-triggers/
event for userA, and event for userB are treated the same, so there no way to trigger separate action for userA or userB right?
![image](https://user-images.githubusercontent.com/47600586/114839628-30bfd900-9e00-11eb-966f-b6dc743764f4.png)

**Reply by thanhpd-teko** [reply]
What I want here is 
- a1 + b1 => trigger action for user 1
- a2 + b2 => trigger action for user 2
- a3 + b3 => trigger action for user 3
- a4 + _ => wait for b4
